indico/modules/events/contributions/controllers.py

```diff
@@ -175,3 +182,61 @@ class RHContributionPersonList(RHContributionPersonListMixin, RHManageContributi
+            self._handle_contrib_managers(form.managers.data)
+            self._handle_protection_mode(form.protection_mode.data)
+            self._handle_access_control_list(form.acl.data)
+            self._handle_contrib_submitters(form.submitters.data)
+            return jsonify_data(flash=False)
+        protection_message = _render_contrib_protection_message(self.contrib)
+        return jsonify_template('events/contributions/management/contribution_protection.html', form=form,
+                                protection_message=protection_message)
+
+    def _get_defaults(self):
+        managers = {p.principal for p in self.contrib.acl_entries if p.full_access}
+        submitters = {p.principal for p in self.contrib.acl_entries if 'submit' in p.roles}
+        acl = {p.principal for p in self.contrib.acl_entries if p.read_access}
+        return {'managers': managers, 'submitters': submitters, 'protection_mode': self.contrib.protection_mode.name,
+                'acl': acl}
+
+    def _handle_contrib_managers(self, managers_data):
+        current_managers = {(acl.principal.name if acl.principal.is_group else acl.principal.id): acl.principal
+                            for acl in self.contrib.acl_entries if acl.full_access}
+        new_managers = {(m.group.name if m.is_group else m.id): m for m in managers_data}
+        deleted_managers_ids = current_managers.viewkeys() - new_managers.viewkeys()
+        for deleted_manager_id in deleted_managers_ids:
+            self.contrib.update_principal(current_managers[deleted_manager_id], full_access=False)
+        for new_manager_id in new_managers.viewkeys() - current_managers.viewkeys():
+            self.contrib.update_principal(new_managers[new_manager_id], full_access=True)
+
+    def _handle_protection_mode(self, protection_mode):
+        if ProtectionMode.get(protection_mode) is not None:
+            self.contrib.protection_mode = ProtectionMode.get(protection_mode)
+
+    def _handle_access_control_list(self, access_control_list):
+        current_viewers = {(acl.principal.name if acl.principal.is_group else acl.principal.id): acl.principal
+                           for acl in self.contrib.acl_entries if acl.read_access}
+        new_viewers = {(m.group.name if m.is_group else m.id): m for m in access_control_list
+                       if self.contrib.protection_mode == ProtectionMode.protected}
+        deleted_viewers_ids = current_viewers.viewkeys() - new_viewers.viewkeys()
+        for deleted_viewer_id in deleted_viewers_ids:
+            self.contrib.update_principal(current_viewers[deleted_viewer_id], read_access=False)
+        for new_viewer_id in new_viewers.viewkeys() - current_viewers.viewkeys():
+            self.contrib.update_principal(new_viewers[new_viewer_id], read_access=True)
+
+    def _handle_contrib_submitters(self, submitters_data):
+        current_submitters = {(acl.principal.name if acl.principal.is_group else acl.principal.id): acl.principal
```

Comment:
same as for managers, just s/full_access/has_management_role('submit')/ and add_roles/del_roles in the acl update calls